### ISSUE TITLE
"Analyzer->Circle Intersection" do wrong result?

### ISSUE THREAD

**Issue by satabol** [nortikin/sverchok#4708]
Windows 11, Blender 3.3.1, Sverchok master.

![image](https://user-images.githubusercontent.com/14288520/196001940-45423085-713f-4af7-87fd-181128e1dd85.png)

http://nortikin.github.io/sverchok/docs/nodes/analyzer/intersect_circle_circle.html

I visualize context of node "Analyzer->Circle Intersection" when Intersection Plane defined by Normal. If parameters are simple then result is good and expectable:

![image](https://user-images.githubusercontent.com/14288520/196001768-ebd485e7-6118-4b32-94fa-2791fb1bc6e8.png)

But if I set "Normal" input in direction when it is not normal to Line AB then I got wrong result:

![image](https://user-images.githubusercontent.com/14288520/196002511-9a386c78-d404-4d67-9bca-dd2f7775d0cb.png)

I found than this appears in cases if "Normal" input is not normal to line OriginA-OriginB.

![Intersection Circle-Circle Error](https://user-images.githubusercontent.com/14288520/196002167-8780e542-6c0a-4f59-996e-6eb2883881ba.gif)

If I align it to normal of Line OriginA-OriginB with additional nodes then result is good:

![Intersection Circle-Circle Error 002](https://user-images.githubusercontent.com/14288520/196002298-f38947f6-156d-4171-a103-710c0bfecae7.gif)

Is protection of Normal parameter is solution for this case or I do some thing wrong?

P.S.

You can test it with:

https://gist.github.com/satabol/667a036e484d04102e35d2f5e6dfad5d#file-analizer-circle-circle-intersection-json

Code node:

https://github.com/nortikin/sverchok/blob/45beb7aee659a141b15842cdd74bab00fb5dd483/nodes/analyzer/intersect_circle_circle.py#L28-L84

**Comment by Durman** [comment]
I guess that non perpendicular normal to the line AB is invalid input. The solution could be to project an invalid normal to perpendicular surface.

**Comment by Durman** [comment]
I guess it can be done simple as that
```py
perpendicular = cross_product(wrong_normal, line_direction)
fixed_normal = cross_product(perpendicular, line_direction)
```
I'm not sure about the order of the arguments.

**Comment by Durman** [comment]
I don't think it would be wise to add sockets which are irrelevant to the function of the node. If the node is named Intersection then it should return intersections. Matrix intersection could be added but it looks redundant because it can be easily constructed from intersection points if someone has such need.

**Comment by satabol** [comment]
@Durman  Yes, you right. But this node is not simple. The name of node says about intersection of circles (and ask about normal? Why?) but in background it use intersection of spheres and a plane. I was embarrassed when I try to use Z axis and got strange result that I cant explain. I think the name of this node describe it functionality partially.  But rename of node is bad.
But this is just idea.
